src/qtbase/QColor.js

```diff
@@ -1,17 +1,208 @@
+    }
+
+    COLOR_STRING_PARSER.some(parser => {
+      const match = parser.re.exec(_color_string);
+      if (match) {
+        const values = parser.parse(match);
+        const spaceName = parser.space || "rgba";
+
+        const props = SPACES[spaceName];
+
+        rgba = [0, 0, 0, 0];
+
+        props.forEach((_prop_type, i) => {
+          rgba[i] = clamp(values[i], _prop_type);
+        });
+
+        // exit each( COLOR_STRING_PARSER ) here because we matched
+        return true;
+      }
+      return false;
+    });
+
+    return rgba || COLORS.unknown; // wrong color :black
+  };
+}());
+
+// [ColorKeywords](https://www.w3.org/TR/SVG/types.html#ColorKeywords)
+(function() {
+  const NAMED_COLORS = "Alicebluef0f8ffAntiquewhitefaebd7Aqua00ffffAquamarine7fffd4Azuref0ffffBeigef5f5dcBisqueffe4c4Black000000BlanchedalmondffebcdBlue0000ffBlueviolet8a2be2Browna52a2aBurlywooddeb887Cadetblue5f9ea0Chartreuse7fff00Chocolated2691eCoralff7f50Cornflowerblue6495edCornsilkfff8dcCrimsondc143cCyan00ffffDarkblue00008bDarkcyan008b8bDarkgoldenrodb8860bDarkgraya9a9a9Darkgreya9a9a9Darkgreen006400Darkkhakibdb76bDarkmagenta8b008bDarkolivegreen556b2fDarkorangeff8c00Darkorchid9932ccDarkred8b0000Darksalmone9967aDarkseagreen8fbc8fDarkslateblue483d8bDarkslategray2f4f4fDarkslategrey2f4f4fDarkturquoise00ced1Darkviolet9400d3Deeppinkff1493Deepskyblue00bfffDimgray696969Dimgrey696969Dodgerblue1e90ffFirebrickb22222Floralwhitefffaf0Forestgreen228b22Fuchsiaff00ffGainsborodcdcdcGhostwhitef8f8ffGoldffd700Goldenroddaa520Gray808080Grey808080Green008000Greenyellowadff2fHoneydewf0fff0Hotpinkff69b4Indianredcd5c5cIndigo4b0082Ivoryfffff0Khakif0e68cLavendere6e6faLavenderblushfff0f5Lawngreen7cfc00LemonchiffonfffacdLightblueadd8e6Lightcoralf08080Lightcyane0ffffLightgoldenrodyellowfafad2Lightgrayd3d3d3Lightgreyd3d3d3Lightgreen90ee90Lightpinkffb6c1Lightsalmonffa07aLightseagreen20b2aaLightskyblue87cefaLightslategray778899Lightslategrey778899Lightsteelblueb0c4deLightyellowffffe0Lime00ff00Limegreen32cd32Linenfaf0e6Magentaff00ffMaroon800000Mediumaquamarine66cdaaMediumblue0000cdMediumorchidba55d3Mediumpurple9370dbMediumseagreen3cb371Mediumslateblue7b68eeMediumspringgreen00fa9aMediumturquoise48d1ccMediumvioletredc71585Midnightblue191970Mintcreamf5fffaMistyroseffe4e1Moccasinffe4b5NavajowhiteffdeadNavy000080Oldlacefdf5e6Olive808000Olivedrab6b8e23Orangeffa500Orangeredff4500Orchidda70d6Palegoldenrodeee8aaPalegreen98fb98PaleturquoiseafeeeePalevioletreddb7093Papayawhipffefd5Peachpuffffdab9Perucd853fPinkffc0cbPlumdda0ddPowderblueb0e0e6Purple800080Rebeccapurple663399Redff0000Rosybrownbc8f8fRoyalblue4169e1Saddlebrown8b4513Salmonfa8072Sandybrownf4a460Seagreen2e8b57Seashellfff5eeSiennaa0522dSilverc0c0c0Skyblue87ceebSlateblue6a5acdSlategray708090Slategrey708090SnowfffafaSpringgreen00ff7fSteelblue4682b4Tand2b48cTeal008080Thistled8bfd8Tomatoff6347Turquoise40e0d0Violetee82eeWheatf5deb3WhiteffffffWhitesmokef5f5f5Yellowffff00Yellowgreen9acd32" // eslint-disable-line max-len
+    .match(/[A-Z][0-9|a-z]+/g);
```

Comment:
There is no reason to do something like this, what's wrong with a regular object?